Automattic/jetpack
# Add/launchpad apis to dismiss page editor modal

Related to https://github.com/Automattic/wp-calypso/issues/76880

## Proposed changes:

* This PR adds a new `hide_fse_next_steps_modal` sub-option to the `wpcom_launchpad_config` option to control whether we show or hide the launchpad modal we show in the editor
* The PR includes an update to the existing `POST /launchpad` API to toggle the value of the option, as well as APIs to get and set the value in `wpcom_launchpad_is_fse_next_steps_modal_hidden()` and `wpcom_launchpad_set_fse_next_steps_modal_hidden()`, respectively.

Note that this PR was split out of https://github.com/Automattic/jetpack/pull/32567 to simplify testing.

### Other information:


## Jetpack product discussion


This changes a WordPress.com-specific feature, so wasn't discussed with the Jetpack team.

## Does this pull request change what data or activity we track or use?


No changes to data or tracking.

## Testing instructions:






* Check this branch out locally, and run the unit tests:
```sh
cd projects/packages/jetpack-mu-wpcom/
vendor/bin/phpunit
```
* Use the instructions in [this comment](https://github.com/Automattic/jetpack/pull/32730#issuecomment-1697226494) to apply the changes in this patch to your WordPress.com development sandbox
* Ensure that `public-api.wordpress.com` is pointed at your sandbox
* Open the [WordPress.com developer console](https://developer.wordpress.com/docs/api/console/), and ensure that you're querying the WP REST API and `wpcom/v2`.
* Update the query to `POST` to `/sites/$wpcom_site/launchpad`
* Verify that you see a `hide_fse_next_steps_modal` field
* Update `$wpcom_site` to your site slug or ID
* Enter `true` for the field, and submit the query
* Verify that the update has been effected (and you can confirm on the back end, too)
* Enter `false` for the field and submit the query
* Verify that the update has been effected
